agent/config/config.go

```diff
@@ -355,6 +357,40 @@ func fileConfig() (Config, error) {
 	return cfg, nil
 }
 
+// userDataConfig reads configuration JSON from instance's userdata. It doesn't
+// return any error as it's entirely optional to configure the ECS agent using
+// this method.
+// Example:
+// {"ECSAgentConfiguration":{"Cluster":"default"}}
+func userDataConfig(ec2Client ec2.EC2MetadataClient) Config {
+	type userDataParser struct {
+		Config Config `json:"ECSAgentConfiguration"`
+	}
+
+	parsedUserData := userDataParser{
+		Config: Config{},
+	}
+
+	userData, err := ec2Client.GetUserData()
+	if err != nil {
```

Comment:
It would have been nice to return an error whenever the UserData format is invalid (i.e. parsing error), instead of ignoring. But I don't know how expensive that is and whether it is worth the effort.